### DISCUSSION TITLE
Chat translate feature

### DISCUSSION THREAD

**Discussion by RoyDubnium** [maruohon/tweakeroo#289] [Ideas]
My idea is for a feature where you can automatically translate incoming and outgoing chat messages independently of each other. For example, you could automatically translate all messages sent by other players into English, while simultaneously translating all outgoing messages to, say, Spanish. With other people's messages, it should probably display the original message along with the translated one, just to avoid mistakes. Currently, the only up-to-date fabric mod for 1.17 that I could find capable of doing this was Wurst client, which... I think you know why that would be suboptimal.

**Comment by maruohon** [comment]
This is a lot more likely to appear as part of my upcoming separate chat related mod, rather than as part of Tweakeroo. I don't think this complex of a feature really fits into Tweakeroo's "lots of small random things" concept, especially since I have that separate chat mod planned (and kinda started already) anyway.

But I also don't have any timeline for when that mod might be coming out, but it will likely be developed to a decent state before I finalize and stabilize the on-going malilib rewrites. But in any case it will come via the 1.12.2 main development code, so it will not come to 1.17+ before all of that rewrite stuff for all the client mods is ready and can be ported forward.
